### ISSUE TITLE
MaxRequestLength not populated properly in JS Server Variables

### ISSUE THREAD

**Issue by matthewcare** [umbraco/Umbraco-CMS#13362]
### Which Umbraco version are you using? (Please write the *exact* version, example: 10.1.0)

11.0.0-rc1.preview.88

### Bug summary

`Umbraco.Sys.ServerVariables.umbracoSettings.maxFileSize` isn't populated, and by extention `RuntimeSettings.MaxRequestLength` is also incorrect unless you specify a value for `Umbraco:CMS:Runtime:MaxRequestLength` in appsettings.

web.config's `maxAllowedContentLength` is also incorrectly mapped when hosting with IIS

### Specifics

In `Umbraco.Web.BackOffice.BackOfficeServerVariables` there is this method

```
private string GetMaxRequestLength()
{
    return _runtimeSettings.MaxRequestLength.HasValue ? _runtimeSettings.MaxRequestLength.Value.ToString() : string.Empty;
}
```

Which returns an empty string if the value is empty.
However in  `ConfigureKestrelServerOptions` a default value is set for the `MaxRequestBodySize` if this value is empty.

There also seems to be inconsistencies if you set `maxAllowedContentLength` in a web.config file when the site is hosted with IIS.

### Steps to reproduce

Type `Umbraco.Sys.ServerVariables.umbracoSettings` into the browser console when in the backoffice and observe that `maxFileSize` is empty if the appsetting `Umbraco:CMS:Runtime:MaxRequestLength` is not specified.

### Expected result / actual result

Expected: 
Umbraco.Sys.ServerVariables.umbracoSettings.maxFileSize should return the max request length regardless of if you set it in appsettings / web.config.

Actual:
Umbraco.Sys.ServerVariables.umbracoSettings.maxFileSize is empty unless specified in appsettings.

**Comment by nul800sebastiaan** [comment]
I seem to recall this is on purpose, as you can not override the server level config (kestrel, IIS, nginx) from the application level AFAIK. It's very confusing, I don't know why we have a config, apart from maybe rejecting files that passed the server level config but we don't want to allow in the application level?

**Comment by matthewcare** [comment]
So the `Umbraco:CMS:Runtime:MaxRequestLength` appsetting is used to populate 
`MultipartBodyLengthLimit` in `ConfigureFormOptions` and defaults to `long.MaxValue` if left blank!
The actual default value from Microsoft is:

```
/// <summary>
/// Default value for <see cref="BufferBodyLengthLimit"/>.
/// Defaults to 134,217,728 bytes, which is 128MB.
/// </summary>
public const int DefaultBufferBodyLengthLimit = 1024 * 1024 * 128;
```

So I don't think setting to `long.MaxValue` is something we should be doing.

It's also used in `ConfigureKestrelServerOptions` to set the `MaxRequestBodySize` and again sets a default value which doesn't match the default Microsoft value.
The comment there is
> // convert from KB to bytes, 52428800 bytes (50 MB) is the same as in the IIS settings

TLDR; Max request size isn't aligned for all hosting platforms, and the method to retrieve the value isn't reliable.

Looking forward to some kind of resolution so I can start on client side validation for file uploads :)

**Comment by AndyButland** [comment]
I'm going to close this one off now so there's no confusion for new contributors looking to resolve (see PR linked above).  This isn't directly relevant for newer versions (server variables is no longer a thing, though there are equivalents).  And I feel the current behaviour is reasonable: 

> If the customer explicitly sets a limit, we enforce it on the client; but if they don't the server - whatever networking and web server setup they have - will accept or reject.
